### ISSUE TITLE
Why bother using diffusion-based formulation, since Lotus-D outperforms Lotus-G on most datasets？

### ISSUE THREAD

**Issue by Boatsure** [EnVision-Research/Lotus#31]
Thanks for your excellent work and responses to ICLR reviews.

I fully understand and approve the important findings of your paper, including the criticality of prior of the sdv2 pre-trained model, the effectiveness of single-step diffusion and x0 prediction. These are valuable exploration for the community.

My question is: For dense prediction tasks, if the reason why other discriminative methods are more advantageous than generative methods before is that the data set is larger, then now we have the comparison between Lotus-G and D. Does this prove that discriminative methods are superior to generative methods in dense prediction tasks? Is it necessary for the community to continue to develop generative models for depth and normal estimation?

**Comment by haodong2000** [comment]
Hi @Boatsure, thanks for your interest!

Here is the answer in my own:

Although in depth/normal prediction, areas like sky, object edges may contain uncertainty, there is no uncertainty in both the training data and benchmarks. To some extend, there may be some "domain gap" between the model design of Lotus-G and our exp. settings. While there is no such "gap" on Lotus-D, which is originally purposed to align with the deterministic nature of dense prediction benchmarks.

Moreover, Lotus-D only offers a single sample, while the output of Lotus-G can be considered as "distribution". Thus, sometimes I think it'd be better to treat these two versions of Lotus as different tasks. But as mentioned, there is no benchmark that has "variance". We can not "directly" evaluate the performance of Lotus-G.

---

I know that my answer maybe wired to many people, but anyway, I am always open for discussion!

Best,
Haodong

**Comment by Boatsure** [comment]
Thanks for your sharing @haodong2000 . These are very interesting statements.
 
Given the different nature of the two kinds of models, it seems that they cannot be compared fairly. 

But in any case, the evaluation of the models cannot avoid result-oriented metrics - and in the current dense prediction task, the result is deterministic. 

I think how to retain the generalization ability from the pixel-level diffusion process while suppressing its probabilistic generation is a problem worth studying.

**Comment by BlingHe** [comment]
Thank you for raising this interesting question. I also share my opinion here and I believe this is still an open problem.

Our method is based on diffusion models, so the natural approach is to strictly follow the diffusion formulation, which is what Lotus-G does. As a result, Lotus-G inherits all the properties of diffusion models, meaning it can be seen as a probabilistic model capable of computing variance and uncertainty. I believe the predicted distribution from a probabilistic model is a valuable output, which is why we have retained this version of Lotus.

On the other hand, Lotus-D does not strictly adhere to the diffusion formulation in order to align better with the deterministic nature of dense prediction tasks. Despite this deviation, it performs better in terms of numerical metrics.

In practice, the choice between the two versions depends on the specific needs. 

Also, due to the difference between the two versions of Lotus, for a more fair comparison, we present Lotus-G alongside other generative methods and Lotus-D with deterministic methods, as shown in Tab. 1 and Tab. 2, respectively.

**Comment by Boatsure** [comment]
The small amount of data required to achieve competitive performance with Lotus is impressive.

Hopefully, one day the output of the probability distribution combined with real-world parameters like focal length can eventually generate accurate metric depth images with absolute distance.
